packages/js/src/dashboard/components/sidebar-recommendations.js

```diff
@@ -13,12 +15,49 @@ const SidebarRecommendations = () => {
 		return null;
 	}
 
+	const [ sidebarPosition, setSidebarPosition ] = useState( {} );
+	const [ scrollY, setScrollY ] = useState( 0 );
+
+	const updatePosition = useCallback( () => {
+		setScrollY( window.scrollY );
+	}, [ setScrollY ] );
+
+	useEffect( () => {
+		if ( ! position?.top ) {
+			return;
+		}
+
+		if ( position.top > scrollY ) {
+			setSidebarPosition( { top: position.top - scrollY } );
+		} else {
+			setSidebarPosition( { top: "64px" } );
+		}
+	}, [ scrollY ] );
+
+	useEffect( () => {
+		setScrollY( window.scrollY );
+		window.addEventListener( "scroll", updatePosition );
+		window.addEventListener( "resize", updatePosition );
+		return () => {
+			window.removeEventListener( "scroll", updatePosition );
+			window.removeEventListener( "resize", updatePosition );
+		};
+	}, [] );
+
 	return (
-		<RecommendationsSidebar>
-			<PremiumUpsellCard link={ premiumLink } linkProps={ premiumUpsellConfig } promotions={ promotions } />
-			<AcademyUpsellCard link={ academyLink } />
-		</RecommendationsSidebar>
+		<div className="xl:yst-max-w-3xl xl:yst-fixed xl:yst-right-8 xl:yst-w-[16rem]" style={ sidebarPosition }>
+			<div className="yst-grid yst-grid-cols-1 sm:yst-grid-cols-2 min-[783px]:yst-grid-cols-1 lg:yst-grid-cols-2 xl:yst-grid-cols-1 yst-gap-4">
+				<PremiumUpsellCard link={ premiumLink } linkProps={ premiumUpsellConfig } promotions={ promotions } />
+				<AcademyUpsellCard link={ academyLink } />
+			</div>
```

Comment:
I changed more things here, and I don't want to usr the RecommendationsSidebar because I think it uses not the optimal usage of styles and grid, but I don't want to increase the scope of this PR.